channels/parallel/client/parallel_main.c

```diff
@@ -327,22 +401,31 @@ int DeviceServiceEntry(PDEVICE_SERVICE_ENTRY_POINTS pEntryPoints)
 
 		parallel->queue = MessageQueue_New(NULL);
 		if (!parallel->queue)
-			goto error_queue;
+		{
+			WLog_ERR(TAG, "MessageQueue_New failed!");
+			error = CHANNEL_RC_NO_MEMORY;
+			goto error_out;
+		}
 
-		pEntryPoints->RegisterDevice(pEntryPoints->devman, (DEVICE*) parallel);
+		if ((error = pEntryPoints->RegisterDevice(pEntryPoints->devman, (DEVICE*) parallel)))
+		{
+			WLog_ERR(TAG, "RegisterDevice failed with error %lu!", error);
+			goto error_out;
+		}
 
-		parallel->thread = CreateThread(NULL, 0, (LPTHREAD_START_ROUTINE) parallel_thread_func, (void*) parallel, 0, NULL);
-		if (!parallel->thread)
-			goto error_thread;
-	}
 
-	return 0;
+		if (!(parallel->thread = CreateThread(NULL, 0, (LPTHREAD_START_ROUTINE) parallel_thread_func, (void*) parallel, 0, NULL)))
+		{
+			WLog_ERR(TAG, "CreateThread failed!");
+			error = ERROR_INTERNAL_ERROR;
+			goto error_out;
+		}
+	}
 
-error_thread:
+	return CHANNEL_RC_OK;
+error_out:
```

Comment:
I think having different lables as before might make sense at least for MessageQueue_Free. If I'm not mistaken MessageQueue_Free won't handle "NULL queues" at the moment.